cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -633,6 +633,103 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
+                keys = Object.keys(groupAccess);
+                if (keys.length === 1) { // should be only one partition key
+                    if (groupAccess.hasOwnProperty(keys[0]) && groupAccess[keys[0]].length > 0) {
+                        // Select the option that has group access, provided there is a specific group within the scheme
+                        this.$('.user-partition-select option[value=' + keys[0] + ']').prop('selected', true);
+                        this.showSelectedDiv(keys[0]);
+                        this.$('#partition-select option:contains("Select a group type")').text(
+                            'All Learners and Staff'
+                        );
+                    }
+                }
+            }
+        },
+
+        getSelectedEnrollmentTrackId: function() {
+            return parseInt(this.$('.user-partition-select').val(), 10);
+        },
+
+        getCheckboxDivs: function() {
+            return $('.user-partition-group-checkboxes').children('div');
+        },
+
+        getSelectedCheckboxesByDivId: function(contentGroupId) {
+            var $checkboxes = $('#' + contentGroupId + '-checkboxes input:checked'),
+                selectedCheckboxValues = [],
+                i;
+            for (i = 0; i < $checkboxes.length; i++) {
+                selectedCheckboxValues.push(parseInt($($checkboxes[i]).val(), 10));
+            }
+            return selectedCheckboxValues;
+        },
+
+        showSelectedDiv: function(contentGroupId) {
+            $('#' + contentGroupId + '-checkboxes').show();
+        },
+
+        hideCheckboxDivs: function() {
+            this.getCheckboxDivs().hide();
+        },
+
+        hasChanges: function() {
+            // compare the group access object retrieved vs the current selection
+            return (JSON.stringify(this.model.get('group_access')) !== JSON.stringify(this.getGroupAccessData()));
+        },
+
+        getSelectedGroupAccess: function() {
+            var selectedEnrollmentTrackId = this.getSelectedEnrollmentTrackId();
+            this.getSelectedCheckboxesByDivId(selectedEnrollmentTrackId);
+        },
+
+        getGroupAccessData: function() {
+            var userPartitionId = this.getSelectedEnrollmentTrackId(),
+                groupAccess = {};
+            if (userPartitionId !== 'none' && !isNaN(userPartitionId)) { // If the selected partition ID is not none
+                groupAccess[userPartitionId] = this.getSelectedCheckboxesByDivId(userPartitionId);
+                return groupAccess;
+            } else {
+                return {};
+            }
+        },
+
+        getRequestData: function() {
+            var metadata = {},
+                groupAccessData = this.getGroupAccessData();
+
+            if (this.hasChanges()) {
+                if (groupAccessData) {
+                    metadata.group_access = groupAccessData;
+                }
+                return {
```

Comment:
To make sure I understand, if the condition at 720 is False, then group access is being cleared (set to All Learners and Staff)?